scene/gui/text_edit.cpp

```diff
@@ -5173,15 +5173,21 @@ void TextEdit::cut() {
 
 	if (!selection.active) {
 
+		auto initial_cursor_line = cursor.line;
```

Comment:
FYI `auto` is banned in Godot.

In any case it should be possible to use `cursor.line` in the below if.

Reply:
ok, thanks! Where can I find more rules (like _forbidden auto_) related to Godot?

Reply:
Currently Godot is still C++03. But, for 4.0 the C++11 guidelines are still being written up. 

If you're keen though, there was an overview given at GodotCon Brussels 2019 that should be on YouTube.